back/src/main/java/com/ubikloadpack/jmeter/ulp/observability/registry/MicrometerRegistry.java

```diff
@@ -124,32 +136,65 @@ public synchronized void addResponse(ResponseResult result) {
 			return;
 		}
 
+		// Create micrometer tags to feed with data
 		String threadTag = Util.makeMicrometerName(result.getThreadGroupLabel());
 		String samplerTag = "spl_"+Util.makeMicrometerName(result.getSamplerLabel());
+		List<String> micrometerTags = Arrays.asList(new String[]{threadTag, samplerTag, this.totalLabel});
 		
-		this.registry.summary("summary.response", "sample", threadTag).record(result.getResponseTime());
-		this.registry.summary("summary.response", "sample", this.totalLabel).record(result.getResponseTime());
-		this.registry.summary("summary.response", "sample", samplerTag).record(result.getResponseTime());
-	
-		this.registry.counter("count.threads", "sample", threadTag).increment(
-				result.getGroupThreads() - (int) this.registry.counter("count.threads", "sample", threadTag).count());
-		this.registry.counter("count.threads", "sample", this.totalLabel).increment(
-				result.getAllThreads() - (int) this.registry.counter("count.threads", "sample", this.totalLabel).count());
-		this.registry.counter("count.threads", "sample", samplerTag).increment(
-				result.getGroupThreads() - (int) this.registry.counter("count.threads", "sample", threadTag).count());
-	
-		if(result.hasError()) {
-			this.registry.counter("count.error", "sample", threadTag).increment();
-			this.registry.counter("count.error", "sample", this.totalLabel).increment();
-			this.registry.counter("count.error", "sample", samplerTag).increment();
+		for(String microMeterTag : micrometerTags) {
+			// Save current period responses time
+			this.registry.summary("summary.response", "sample", microMeterTag).record(result.getResponseTime());
```

Comment:
for clarity can you rename registry to intervalRegistry
and totalReg to summaryRegistry 

Thanks